docs/nddata/subclassing.rst

```diff
@@ -1,100 +1,488 @@
+    ...         # Call the setter of the super class in case it might contain some
+    ...         # important logic (only True for meta, unit and uncertainty)
+    ...         super(NDDataUncertaintyShapeChecker, self.__class__).uncertainty.fset(self, value)
+
+    >>> ndd = NDDataUncertaintyShapeChecker([1,2,3], uncertainty=[2,3,4])
+    INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
+    >>> ndd.uncertainty
+    UnknownUncertainty([2, 3, 4])
+
+Having a setter for the data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    >>> class NDDataWithDataSetter(NDData):
+    ...
+    ...     @NDData.data.setter
+    ...     def data(self, value):
+    ...         # Convert mask to numpy array
+    ...         self._data = np.asarray(value)
+
+    >>> ndd = NDDataWithDataSetter([1,2,3])
+    >>> ndd.data = [3,2,1]
+    >>> ndd.data
+    array([3, 2, 1])
+
+`~astropy.nddata.NDDataRef`
+---------------------------
+
+`~astropy.nddata.NDDataRef` itself inherits from `~astropy.nddata.NDData` so
+any of the possibilities there also apply to NDDataRef. But NDDataRef also
+inherits from the Mixins:
+
+- `~astropy.nddata.NDSlicingMixin`
+- `~astropy.nddata.NDArithmeticMixin`
+- `~astropy.nddata.NDIOMixin`
+
+which allow additional operations.
+
+
+Arithmetic on an existing property
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Customizing how an existing property is handled during arithmetics is possible
+with some arguments to the function calls like
+:meth:`~astropy.nddata.NDArithmeticMixin.add` but if it's possible to hardcode
+behaviour too. The actual operation on the attribute (except for ``unit``) is
+done in a method ``_arithmetic_*`` where ``*`` is the name of the property.
+
+For example to customize how the ``meta`` will be affected during arithmetics::
+
+    >>> from astropy.nddata import NDDataRef
+
+    >>> from copy import deepcopy
+    >>> class NDDataWithMetaArithmetics(NDDataRef):
+    ...
+    ...     def _arithmetic_meta(self, operation, operand, handle_mask, **kwds):
+    ...         # the function must take the arguments:
+    ...         # operation (numpy-ufunc like np.add, np.subtract, ...)
+    ...         # operand (the other NDData-like object, already wrapped as NDData)
+    ...         # handle_mask (see description for "add")
+    ...
+    ...         # The meta is dict like but we want the keywords exposure to change
+    ...         # Anticipate that one or both might have no meta and take the first one that has
+    ...         result_meta = deepcopy(self.meta) if self.meta else deepcopy(operand.meta)
+    ...         # Do the operation on the keyword if the keyword exists
+    ...         if result_meta and 'exposure' in result_meta:
+    ...             result_meta['exposure'] = operation(result_meta['exposure'], operand.data)
+    ...         return result_meta # return it
+
+To trigger this method the ``handle_meta`` argument must not be ``None``,
+``"ff"`` or ``"first_found"``::
```

Comment:
Could this be phrased in terms of what it *should* be set to, so something like: 

To trigger this method the ``handle_meta`` argument to arithmetic methods can be anything except ``None`` or ``"first_found"``::